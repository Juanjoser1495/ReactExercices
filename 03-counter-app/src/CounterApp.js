import PropTypes from "prop-types";
import { useState } from "react";

const CounterApp = ({ value = 0 }) => {
  const [counter, setCounter] = useState(value);

  //handleAdd
  const handleAdd = (e) => {
    console.log(e);
  };

  //handleAdd use State
  const handleAddState = () => {
    setCounter(counter + 1);
  };

  //handle Reset
  const handleReset = () => setCounter(value);
  //handle substract
  const handleSubstract = () => {
    setCounter((c) => {
      return c - 1;
    });
  };

  return (
    <div>
      <h1>CounterApp</h1>
      <h2>{value}</h2>
      <h3>{counter}</h3>
      <button
        onClick={(e) => {
          handleAdd(e);
        }}
      >
        +1
      </button>
      <button onClick={handleAddState}>+1 use state</button>
      <button onClick={handleReset}>Reset</button>
      <button onClick={handleSubstract}>-1 use state</button>
    </div>
  );
};

CounterApp.propTypes = {
  value: PropTypes.number,
};

export default CounterApp;
