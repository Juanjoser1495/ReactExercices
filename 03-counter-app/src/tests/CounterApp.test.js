import { shallow } from "enzyme";
import CounterApp from "../CounterApp";

describe("Pruebas en <CounterApp />", () => {
  test("Debe de mostrar el <CounterApp> correctamnete", () => {
    const wrapper = shallow(<CounterApp></CounterApp>);
    expect(wrapper).toMatchSnapshot();
  });

  test("Debe de mostrar el value enviado por props", () => {
    const value = "100";
    const wrapper = shallow(<CounterApp value={100}></CounterApp>);
    const returned = wrapper.find("h3").text();
    expect(returned).toBe(value);
  });
});
