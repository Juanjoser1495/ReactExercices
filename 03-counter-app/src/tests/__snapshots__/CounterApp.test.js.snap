// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <CounterApp /> Debe de mostrar el <CounterApp> correctamnete 1`] = `
<div>
  <h1>
    CounterApp
  </h1>
  <h2>
    0
  </h2>
  <h3>
    0
  </h3>
  <button
    onClick={[Function]}
  >
    +1
  </button>
  <button
    onClick={[Function]}
  >
    +1 use state
  </button>
  <button
    onClick={[Function]}
  >
    Reset
  </button>
  <button
    onClick={[Function]}
  >
    -1 use state
  </button>
</div>
`;
